generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -802,4 +804,60 @@ public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
 
   }
 
+  private class IdentityRewriteComposite extends AbstractIdentityTransformation {
+    private IdentityRewriteComposite() {
+      // all non-boolean vector/matrix types
+      super(BasicType.allNumericTypes().stream().filter(
+          item -> !BasicType.allScalarTypes().contains(item)).collect(Collectors.toList()),
+          false);
+    }
+
+    @Override
+    public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
+                      Fuzzer fuzzer) {
+      // v -> (true ? vecX(..., identity(v[Y]), ...) : _)
+      // v -> (false ? _ : vecX(..., identity(v[Y]), ...))
+      // where v is a vector of size X, y is the random entry in v we want to apply
+      // identities to, and ... is the other entries in v that we don't change.
+      // Similarly for matrices.
+
+      assert BasicType.allVectorTypes().contains(type)
+          || BasicType.allMatrixTypes().contains(type);
+
+      final int numColumns =
+          (BasicType.allVectorTypes().contains(type)
+          ? type.getNumElements() : type.getNumColumns());
+      final int indexToApplyIdentities = generator.nextInt(numColumns);
+      final List<Expr> nonscalarEntryList = new ArrayList<>();
+      for (int i = 0; i < numColumns; i++) {
+        // v + vec2(0.0)[0]; is not what we want to do, so we wrap expr into
```

Comment:
I had an original comment: Please rewrite comment to be a bit more descriptive.  E.g.: "If we are transforming an expression that already has the form 'v + vec2(0.0)', we would not want to generate an expression of the form 'v + vec2(0.0)[0]' to index into this; we would instead want it to look like '(v + vec2(0.0))[0]' ..."

But it then occurred to me that since we limit expr to being a VariableIdentifierExpr, we don't have this problem and do not need to add parentheses at all, so please simplify the code accordingly.